File: scala/sdktest/InitEntityTest.scala
// Generated basic-flow test for the init entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped InitTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefinpayconexsdk.core.{Helpers, BluefinPayconexSDK}
import voxgig.bluefinpayconexsdk.utility.struct.Struct

object InitEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("init.instance") {
      val testsdk = BluefinPayconexSDK.testSDK()
      val ent = testsdk.init(null)
      rep.check("init.instance", ent != null, "expected non-null init entity")
    }

    rep.scope("init.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/init/InitTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinPayconexSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("init01", "INIT01")
      idmap.put("init02", "INIT02")
      idmap.put("init03", "INIT03")
      idmap.put("account01", "ACCOUNT01")
      idmap.put("account02", "ACCOUNT02")
      idmap.put("account03", "ACCOUNT03")
      val now = System.currentTimeMillis()

      // CREATE
      val initRef01Ent = client.init(null)
      var initRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.init"), "init_ref01"))
      initRef01Data.put("account_id", idmap.get("account01"))
      val initRef01DataResult = initRef01Ent.create(initRef01Data, null)
      initRef01Data = Helpers.toMapAny(initRef01DataResult)
      rep.check("init.create.map", initRef01Data != null, "expected create result to be a map")
    }
  }
}

File: elixir/test/init_entity_test.exs
# Init entity test (offline, mock transport)

defmodule BluefinPayconex.InitEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinPayconex.Helpers, as: H
  alias BluefinPayconex.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/init/InitTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinPayconex.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.init"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinPayconex.test()
    ent = BluefinPayconex.init(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = BluefinPayconex.test(S.jm(["entity", S.jm(["init", S.jm([])])]))
    ent = BluefinPayconex.init(sdk)
    made = BluefinPayconex.Entity.Init.create(ent, S.jm(["name", "test-create"]))
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end
